test/groovy/AbapEnvironmentPullGitRepoTest.groovy

```diff
@@ -141,24 +141,37 @@ public class AbapEnvironmentPullGitRepoTest extends BasePiperTest {
         thrown.expect(Exception)
         thrown.expectMessage("[abapEnvironmentPullGitRepo] Error: 401 Unauthorized")
 
-        stepRule.step.abapEnvironmentPullGitRepo(script: nullScript, host: 'example.com', repositoryName: 'Z_DEMO_DM', credentialsId: 'test_credentialsId')
+        stepRule.step.abapEnvironmentPullGitRepo(script: nullScript, url: 'https://1234-abcd-5678-efgh-ijk.abap.eu10.hana.ondemand.com/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY', repositoryName: '/DMO/REPO', credentialsId: 'test_credentialsId')
 
     }
 
     @Test
     public void checkRepositoryProvided() {
        thrown.expect(IllegalArgumentException)
        thrown.expectMessage("Repository / Software Component not provided")
-       stepRule.step.abapEnvironmentPullGitRepo(script: nullScript, host: 'example.com', credentialsId: 'test_credentialsId')
+       stepRule.step.abapEnvironmentPullGitRepo(script: nullScript, url: 'https://1234-abcd-5678-efgh-ijk.abap.eu10.hana.ondemand.com/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY', credentialsId: 'test_credentialsId')
     }
 
     @Test
-    public void checkHostProvided() {
+    public void checkUrlProvided() {
        thrown.expect(IllegalArgumentException)
-       thrown.expectMessage("Host not provided")
+       thrown.expectMessage("URL not provided")
        stepRule.step.abapEnvironmentPullGitRepo(script: nullScript, repositoryName: 'REPO', credentialsId: 'test_credentialsId')
     }
 
+    @Test
+    public void rejectHttpUrl() {
+        thrown.expect(Exception)
+        thrown.expectMessage("[abapEnvironmentPullGitRepo] Error: Please provide a valid URL")
+        stepRule.step.abapEnvironmentPullGitRepo(script: nullScript, url: 'http://1234-abcd-5678-efgh-ijk.abap.eu10.hana.ondemand.com/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY', repositoryName: '/DMO/REPO', credentialsId: 'test_credentialsId')
```

Comment:
I would like to propose to provide a reason why the url is rejected. The url is not semantically invalid. Maybe by providing the regex we use for the check.

Reply:
done